### ISSUE TITLE
The widget single cell scrolling methods no longer work as expected from the keyboard

### ISSUE THREAD

**Issue by davep** [Textualize/textual#1897]
Consider the following (slightly over-complicated, but designed to show all the related information) code:

```python
from rich.segment        import Segment
from textual.app         import App, ComposeResult
from textual.widgets     import Header, Footer
from textual.scroll_view import ScrollView
from textual.geometry    import Size
from textual.strip       import Strip

class ScrollViewTest( ScrollView, can_focus=True ):

    def __init__( self, height: int ) -> None:
        super().__init__()
        self.virtual_size = Size( 0, height )

    def render_line( self, y: int ) -> Strip:
        return Strip( [
            Segment( f"{self.scroll_offset.y + y:10}" ),
            Segment( f"{y:10}" ),
            *(
                [ Segment( f"{' ' * 10}scroll_offset.y == {self.scroll_offset.y}") ]
                if y == 0 else []
            )
        ] )

class ScrollTestApp( App[ None ] ):

    def compose( self ) -> ComposeResult:
        yield Header()
        yield ScrollViewTest( 1000 )
        yield Footer()

    def on_mount( self ) -> None:
        self.query_one( ScrollViewTest ).focus()

if __name__ == "__main__":
    ScrollTestApp().run()
```

If you run it and, using just the keyboard, try and scroll down just one line, or up one line, you'll find it's not possible. It will always scroll two lines.

It seems that this was introduced as part of #1689 ("Scroll sensitivity"). Two `App`-level properties were introduced, called `scroll_sensitivity_x` and `scroll_sensitivity_y`. Both properties say they dictate the number of cells to scroll in the given direction *"with wheel or trackpad"*. However, they're used in the related `Widget.scroll_*` methods regardless of the input method.

At the very least `scroll_up`, `scroll_down`, `scroll_left` and `scroll_right` should be modified to only apply the sensitivity setting if the input device is a pointing device rather than a key binding.

**Comment by davep** [comment]
Bumping this to TODO and claiming it as it intersects with what'll need doing with #1774. This should be merged first I feel, then #1774 can follow (or it may be done as part of the same change; I've yet to decide -- either way I'm promoting it and claiming it).

**Comment by willmcgugan** [comment]
I think we need to differentiate between mouse scroll (which *should* respect sensitivity and keyboard scroll which should *not* respect sensitivity). Suspect we will need to add a "mouse_initiated" bool to the scroll up / down / left / right methods.
